crates/shared/src/zeroex_api.rs

```diff
@@ -260,8 +261,8 @@ pub struct Order {
 #[derive(Debug, Default, Clone, Deserialize, Serialize, Eq, PartialEq)]
 pub struct OrderRecord {
     #[serde(rename = "metaData")]
-    pub metadata: OrderMetadata,
-    pub order: Order,
+    pub metadata: Arc<OrderMetadata>,
+    pub order: Arc<Order>,
```

Comment:
These struct only make sense together so they should be stored inside the same `Arc`.